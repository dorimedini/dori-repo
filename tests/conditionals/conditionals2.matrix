//Check Matrix In Operation
{
	int x;
	matrix m(2,2);
	if(0 in m){
		print "Im happy";
	}
	m[0,0]=1;
        if(x+x*x+0 in m){
        	print "Still happy..";
        }
	m[0,1]=1;
	m[1,0]=1;
	m[1,1]=-1;
        if(0 in m){        
                print "Sad.";
        }
	if(-m[0,0] in m){
                print "Still happy...";
        }

	matrix m1(1,1) = [1],m2(1,2) = [1,2];
        if(m[0,0] in m1 && (m[0,0] in m2)){  
                print "Still happy...."; 
        }
	
	if(1 in m1){
		if(!5 in m1)
			print "Super Happy!";
	}
}
