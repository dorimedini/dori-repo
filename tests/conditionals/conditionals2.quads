s[0]=0
s[1]=0
s[2]=0
s[3]=0
s[4]=0
if s[1]==0 goto 10
if s[2]==0 goto 10
if s[3]==0 goto 10
if s[4]==0 goto 10
goto 11
printLn "Im happy"
s[1]=1
s[5]=s[0]*s[0]
s[6]=s[0]+s[5]
s[7]=s[6]+0
if s[1]==s[7] goto 20
if s[2]==s[7] goto 20
if s[3]==s[7] goto 20
if s[4]==s[7] goto 20
goto 21
printLn "Still happy.."
s[2]=1
s[3]=1
s[8]=0
s[9]=s[8]-1
s[4]=s[9]
if s[1]==0 goto 31
if s[2]==0 goto 31
if s[3]==0 goto 31
if s[4]==0 goto 31
goto 32
printLn "Sad."
s[10]=0
s[11]=s[10]-s[1]
if s[1]==s[11] goto 39
if s[2]==s[11] goto 39
if s[3]==s[11] goto 39
if s[4]==s[11] goto 39
goto 40
printLn "Still happy..."
s[12]=1
s[13]=s[12]
s[14]=1
s[15]=2
s[16]=s[14]
s[17]=s[15]
if s[13]==s[1] goto 48
goto 52
if s[16]==s[1] goto 51
if s[17]==s[1] goto 51
goto 52
printLn "Still happy...."
if s[13]==1 goto 54
goto 57
if s[13]==5 goto 57
goto 56
printLn "Super Happy!"
