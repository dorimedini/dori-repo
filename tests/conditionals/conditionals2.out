Im happy
Still happy..
Still happy...
Still happy....
Super Happy!
